### PR TITLE
Fixes #32362 - Change default pulp url to primary pulp smart proxy url

### PR DESCRIPTION
**Pull request by rverdile**
Also moved full_path method from repo.rb to repository.rb.

### CODE REVIEW

**Comment by ianballou** [comment]
[test katello]

File: app/models/katello/repository.rb [hunk 1]
```diff
@@ -399,6 +399,20 @@ def group
       all_instances
     end
 
+    def full_path(smart_proxy = nil, force_http = false)
```
Comment (ianballou) [inline app/models/katello/repository.rb:402]: Can you add a test to this file? https://github.com/Katello/katello/blob/master/test/models/repository_test.rb

Your test could look like the following:
```ruby
def test_full_path_defaults_to_pulp_primary
  SETTINGS[:katello][:pulp].update(url: nil)  # Ensure the pulp url is not configured
  assert_equals SmartProxy.pulp_primary.url, <some substring of @repo.full_path>
end
```

**Review by ianballou — COMMENTED** [review]
Looks good to me! Let's just add a test:

**Comment by ianballou** [comment]
[test katello]

File: app/models/katello/repository.rb (hunk 1 above)
Comment (ianballou) [reply app/models/katello/repository.rb:402]: You can run your test with the following command: 
```
ktest test/models/repository_test.rb -ntest_full_path_defaults_to_pulp_primary
```

If you need to debug your test, you can add `require 'pry'` to the top of the test file and stick `binding.pry` as a debugger breakpoint wherever you need it.

**Comment by ianballou** [comment]
[test katello]

**Review by ianballou — APPROVED** [review]
Looks good to me! We just need to wait for all the tests to pass.

**Comment by ianballou** [comment]
@rverdile it looks like you have a RuboCop linting error. You can run rubocop from the Foreman directory with:
```
bundle exec rake katello:rubocop
```

**Comment by rverdile** [comment]
> @rverdile it looks like you have a RuboCop linting error. You can run rubocop from the Foreman directory with:
> 
> ```
> bundle exec rake katello:rubocop
> ```

@ianballou Thanks. I fixed the offender.

**Comment by ianballou** [comment]
[test katello]

**Comment by ianballou** [comment]
@rverdile so it looks like  the changes here broke some Pulp 2 - related tests.  Fortunately we should be able to just delete everything since it's all Pulp 2 related.

For example, you can try deleting the method: `spec/models/model_spec_helper.rb:138:in `disable_consumer_group_orchestration'` and remove its usages from all of the tests that failed. Then do the same thing for the other methods that are causing troubles (like `disable_product_orchestration`)

**Comment by ianballou** [comment]
[test katello]

**Comment by ianballou** [comment]
[test katello]

**Review by ianballou — COMMENTED** [review]
Looking good, just one thing to remove:

**Comment by ianballou** [comment]
[test katello]

**Comment by ianballou** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by ianballou** [comment]
I found a couple more usages of the old pulp url setting. You can find them with `grep -nr "SETTINGS\[:katello\]\[:pulp\]\[:url\]"`:
```
vagrant@centos7-katello-devel ~/katello $ grep -nr "SETTINGS\[:katello\]\[:pulp\]\[:url\]"
app/models/katello/glue/pulp/repo.rb:77:        uri = URI.parse(SETTINGS[:katello][:pulp][:url])
test/support/vcr.rb:86:      uri = URI.parse(SETTINGS[:katello][:pulp][:url])
```

You can delete the entire block in the VCR test there.  Also, for the pulp repo glue bit, you should be able to delete the entire method that's using the pulp url.

**Pull request merged**